src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -145,36 +115,17 @@
     public Path[] locations();
 
     /**
-     * Returns the translog file with the given id as a Path. This
-     * will return a relative path.
-     */
-    Path getPath(long translogId);
-
-    /**
      * return stats
      */
     TranslogStats stats();
 
-    /**
-     * Returns the largest translog id present in all locations or <tt>-1</tt> if no translog is present.
+    /** notifies the translog that translogId was committed into lucene, allowing it
+     * to release all previous translogs
```

Comment:
data from 5 & 4 is guaranteed to be part of lucene and the lucene commit point points to 6 for the rest. Data from 6 may or may not be in lucene. I clarified the comment